ui/admin/app/routes/onboarding.js

```diff
@@ -1,36 +1,146 @@
   }
 
   /**
    * List all scopes (orgs) under global.
+   * Creare org, project, target and roles
    */
-  model() {
-    return this.store.query('scope', {
-      scope_id: 'global',
-    });
+  async model() {
+    return {
+      org: this.store.createRecord('scope', {
+        type: 'org',
+        scopeID: 'global',
+        name: 'SecOps',
+        description: 'Sample org created by quick setup',
+      }),
+      project: this.store.createRecord('scope', {
+        type: 'project',
+        name: 'AWS',
+        description: 'Sample project created by quick setup',
+      }),
+      target: this.store.createRecord('target', {
+        type: TYPE_TARGET_TCP,
+        name: 'Test target',
+        description: 'Sample target created by quick setup',
+      }),
+      role: this.store.createRecord('role', {
+        name: 'test_target_role',
+      }),
+    };
+  }
+
+  // =actions
+  @action
+  @loading
+  @notifyError(function () {
+    return this.intl.t('errors.onboarding-failed.description');
+  })
+  async createResources(targetAddress, targetPort) {
+    const { org, project, target, role } = this.currentModel;
+
+    await this.createOrg(org);
+    await this.createProject(org, project);
+    await this.createTarget(project, target, targetAddress, targetPort);
+    await this.createGrants(org, project, target, role);
+
+    // Redirect to success
+    this.router.transitionTo('onboarding.success');
   }
 
   /**
-   * If any orgs already exist, redirect to index to skip onboarding.
+   * Creates Org, if fails, redirects to Orgs empty list
+   * @param {Object} org
    */
-  redirect(model) {
-    // If there are orgs already created, redirect to index.
-    if (model.length) this.router.transitionTo('index');
+  async createOrg(org) {
+    try {
+      await org.save();
+    } catch (error) {
+      // Redirects to Orgs screen
+      this.router.transitionTo('scopes.scope', 'global');
+      throw new Error(error);
+    }
+  }
+
+  /**
+   * Creates Project, if fails redirects to Projects from Org empty list.
+   * @param {Object} org
+   * @param {Object} project
+   */
+  async createProject(org, project) {
+    project.scopeID = org.id;
+
+    try {
+      await project.save();
+    } catch (error) {
+      // Redirects to Project screen
+      this.router.transitionTo('scopes.scope.scopes', project.scopeID);
+      throw new Error(error);
+    }
+  }
+
+  /**
+   * Creates a Target, if fails redirects to Targets empyt list.
+   * @param {Object} project
+   * @param {Object} target
+   * @param {Object} targetAddress
+   * @param {Object} targetPort
+   */
+  async createTarget(project, target, targetAddress, targetPort) {
+    target.scopeID = project.id;
+    target.address = targetAddress;
+    target.default_port = targetPort;
+
+    try {
+      await target.save();
+    } catch (error) {
+      // Redirect to targets global
+      this.router.replaceWith('scopes.scope.targets', 'global');
```

Comment:
Should this be global, wouldn't it be the project scope?

Reply:
Yes, you are right. Changed 😃